src/components/core/pivotTable/ResizedColumnsMap.ts

```diff
@@ -0,0 +1,183 @@
+
+        column.getColDef().suppressSizeToFit = true;
+    }
+
+    public addAllMeasureColumns(columnWidth: number, allColumns: Column[]) {
+        this.allMeasureColumnWidth = columnWidth;
+        allColumns.forEach(col => {
+            if (isMeasureColumn(col)) {
+                const colId = getColumnIdentifier(col);
+                if (this.manuallyResizedColumns[colId]) {
+                    this.manuallyResizedColumns = omit(this.manuallyResizedColumns, colId);
+                }
+                col.getColDef().suppressSizeToFit = true;
+            }
+        });
+    }
+
+    public removeAllMeasureColumns() {
+        this.allMeasureColumnWidth = null;
+        const colIds = Object.keys(this.manuallyResizedColumns);
+        colIds.forEach(colId => {
+            const item = this.manuallyResizedColumns[colId];
+            if (isColumnWidthAuto(item.width)) {
+                this.manuallyResizedColumns = omit(this.manuallyResizedColumns, colId);
+            }
+        });
+    }
+
+    public removeFromManuallyResizedColumn(column: Column): void {
+        const colId = getColumnIdentifier(column);
+        const item = this.manuallyResizedColumns[colId];
+
+        if (item) {
+            this.manuallyResizedColumns = omit(this.manuallyResizedColumns, colId);
+
+            if (!this.isAllMeasureColumWidthUsed()) {
+                column.getColDef().suppressSizeToFit = false;
+            }
+        } else {
+            if (this.isAllMeasureColumWidthUsed() && isMeasureColumn(column)) {
+                this.manuallyResizedColumns[colId] = this.getAutoSizeItem();
+                column.getColDef().suppressSizeToFit = false;
+            }
+        }
+    }
+
+    public getColumnWidthsFromMap(execution: Execution.IExecutionResponses): ColumnWidthItem[] {
+        const result = getColumnWidthsFromMap(this.manuallyResizedColumns, execution);
+        if (this.isAllMeasureColumWidthUsed()) {
+            result.push(this.getAllMeasureColumnWidth());
+        }
+        return result;
+    }
+
+    public updateColumnWidths(
+        columnWidths: ColumnWidthItem[],
+        executionResponse: Execution.IExecutionResponse,
+    ) {
+        const [allMeasureColumnWidthItems, columnWidthItems] = this.partition(columnWidths, item =>
+            isAllMeasureColumnWidthItem(item),
+        );
+
+        const allMeasureWidthItem = allMeasureColumnWidthItems[0];
+
+        if (isAllMeasureColumnWidthItem(allMeasureWidthItem)) {
+            this.allMeasureColumnWidth = allMeasureWidthItem.measureColumnWidthItem.width;
+        } else {
+            this.allMeasureColumnWidth = null;
+        }
+
+        const columnWidthsByField = convertColumnWidthsToMap(columnWidthItems, executionResponse);
+        this.manuallyResizedColumns = columnWidthsByField;
+    }
+
+    private partition(
+        array: ColumnWidthItem[],
+        isValid: (elem: ColumnWidthItem) => boolean,
+    ): [ColumnWidthItem[], ColumnWidthItem[]] {
```

Comment:
I would maybe create some reasonable type instead of array of arrays

Reply:
Skipped